### DISCUSSION TITLE
^ vs ~ in git HEAD

### DISCUSSION THREAD

**Discussion by tbhaxor** [community/community#22165] [New to GitHub]
<p>I know HEAD currently points to the latest commit in current branch.</p>
<p>I have seen many online blogs some of them are using <code>git reset  --soft HEAD~1</code> and <code>git reset --soft HEAD^1</code> to reset the codebase</p>
<p>I checked ~ and ^ has no difference. Please help me understand why git support ^ and ~ if they are same</p>

**Comment by mattmccutchen** [comment, marked as answer]
<p>The reference for commit expression operators such as <code>^</code> and <code>~</code> is <a href="https://git-scm.com/docs/git-rev-parse#_specifying_revisions" rel="noopener nofollow ugc">the “Specifying Revisions” section of the <code>git rev-parse</code> man page</a>.  In general, <code>commit^N</code> means the Nth parent of <code>commit</code>, while <code>commit~N</code> means the commit reached by following the first parent N times.  When N = 1, these are coincidentally the same.  But when N is not 1, they are different: for example, <code>commit^2</code> is the second parent of <code>commit</code>, while <code>commit~2</code> is the first parent of the first parent of <code>commit</code>.</p>
